openedx/features/course_experience/__init__.py

```diff
@@ -104,5 +105,15 @@ def course_home_url(course_key):
     Arguments:
         course_key (CourseKey): The course key for which the home url is being requested.
     """
+    try:
+        # .. filter_implemented_name: CourseHomeUrlCreationStarted
+        # .. filter_type: org.openedx.learning.course.homepage.url.creation.started.v1
+        course_key, course_home_url = CourseHomeUrlCreationStarted.run_filter(
+            course_key=course_key, course_home_url=None
+        )
+        return course_home_url
+    except CourseHomeUrlCreationStarted.PreventCourseHomeUrlRewrite as exc:
```

Comment:
I'm curious about this exception. In other filters an exception has been used to allow for other kinds of behaviours. A good example is to [render a custom page or redirect somewhere else](https://github.com/openedx/edx-platform/blob/aa7370c773dcd96850fbda8e4f6ad40950c10547/lms/djangoapps/courseware/views/views.py#L880). If we are only having the exception to have something to catch in the try block, we could use `OpenEdxFilterException`.

This function is only used for returning a string so actually raising a redirection here does not make much sense. However we could approach this in a slightly different way. Make the filter for the `get_learning_mfe_home_url` call.

In here, it would look like:

```python
try:
    # .. filter_implemented_name: CourseHomeUrlCreationStarted
    # .. filter_type: org.openedx.learning.course.homepage.url.creation.started.v1
    from .url_helpers import get_learning_mfe_home_url
    course_home_url = get_learning_mfe_home_url(course_key, url_fragment='home')

    course_key, course_home_url = CourseHomeUrlCreationStarted.run_filter(
        course_key=course_key, course_home_url=course_home_url
    )
    return course_home_url
except OpenEdxFilterException as exc:
        pass
```

This way the filter pipeline would also get access to the value of `get_learning_mfe_home_url` in case you would want to use that in your deciding logic or to add queryparams to the MFE url without having to know anything about that.


A final note is that [at least ](https://github.com/search?q=repo%3Aopenedx%2Fedx-platform%20course_home_url&type=code)one call to the `course_home_url` string was not calling this function but re-doing the same logic of calling [get_learning_mfe_home_url](https://github.com/openedx/edx-platform/blob/aa7370c773dcd96850fbda8e4f6ad40950c10547/lms/djangoapps/course_goals/management/commands/goal_reminder_email.py#L56-L79).
I would suggest refactoring that send_ace_message to import form course_experience so that this filter is also applied there.